### ISSUE TITLE
RecursionError: maximum recursion depth exceeded while calling a Python object

### ISSUE THREAD

**Issue by krnick** [quark-engine/quark-engine#46]
It seems that the problem of recursive maximum still exists in find_previous_method and upperfunc.

```
Traceback (most recent call last):
  File "/Users/nick/Desktop/quark-engine/quark/Objects/xrule.py", line 71, in find_previous_method
    item, top_method, pre_method_list)
  File "/Users/nick/Desktop/quark-engine/quark/Objects/xrule.py", line 71, in find_previous_method
    item, top_method, pre_method_list)
  File "/Users/nick/Desktop/quark-engine/quark/Objects/xrule.py", line 71, in find_previous_method
    item, top_method, pre_method_list)
  [Previous line repeated 989 more times]
  File "/Users/nick/Desktop/quark-engine/quark/Objects/xrule.py", line 59, in find_previous_method
    method_set = self.apkinfo.upperfunc(class_name, method_name)
  File "/Users/nick/Desktop/quark-engine/quark/Objects/apkinfo.py", line 60, in upperfunc
    method_set = self.find_method(class_name, method_name)
  File "/Users/nick/Desktop/quark-engine/quark/Objects/apkinfo.py", line 44, in find_method
    if len(list(result)) > 0:
  File "/Users/nick/.local/share/virtualenvs/quark-engine-NJwlz3ey/src/androguard/androguard/core/analysis/analysis.py", line 1776, in find_methods
    classname = bytes(mutf8.MUTF8String.from_str(classname))
  File "/Users/nick/.local/share/virtualenvs/quark-engine-NJwlz3ey/src/androguard/androguard/core/mutf8.py", line 102, in from_str
    c = cls(encode(s))
  File "/Users/nick/.local/share/virtualenvs/quark-engine-NJwlz3ey/src/androguard/androguard/core/mutf8.py", line 64, in encode
    ord_array = [i for i in map(lambda x: ord(x), s)]
  File "/Users/nick/.local/share/virtualenvs/quark-engine-NJwlz3ey/src/androguard/androguard/core/mutf8.py", line 64, in <listcomp>
    ord_array = [i for i in map(lambda x: ord(x), s)]
  File "/Users/nick/.local/share/virtualenvs/quark-engine-NJwlz3ey/src/androguard/androguard/core/mutf8.py", line 64, in <lambda>
    ord_array = [i for i in map(lambda x: ord(x), s)]
RecursionError: maximum recursion depth exceeded while calling a Python object
```

I will fix it next week.

sample apk:
https://drive.google.com/file/d/1RqRq_EySvArUuo5OtS-rKNNV3q1H84gL/view?usp=sharing

**Comment by sidra-asa** [comment]
Is this issue identical to #18 ?
The fix is not working?

**Comment by krnick** [comment]
@sidra-asa  I haven't checked the full error log, I just encountered this problem with a big size apk (99 mb), maybe there are other problems.

**Comment by sidra-asa** [comment]
I tested with the sample. I found there's a logical loop if the methods call each other in the method_set. Maybe the previous fix to #18 still needs to be enhanced.

```
Rule: quark/rules/readBookmarks.json

base_method: ('Landroid/content/ContentResolver', 'query'), 
top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig), 
item: (Lcom/amap/location/offline/b;, a), 
method_set: [(Lcom/amap/location/offline/b;, a), (Lcom/amap/api/services/a/bx;, b), (Lcom/amap/api/mapcore/util/hm;, b)]

base_method: ('Landroid/content/ContentResolver', 'query'), 
top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig), 
item: (Lcom/amap/api/services/a/bx;, b), 
method_set: [(Lcom/amap/location/offline/b;, a), (Lcom/amap/api/services/a/bx;, b), (Lcom/amap/api/mapcore/util/hm;, b)]

base_method: (Lcom/amap/api/services/a/bx;, b), top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig), item: (Lcom/amap/api/services/a/bx;, a), method_set: [(Lcom/amap/api/services/a/bx;, a), (Lcom/amap/api/services/a/bx;, b)]

base_method: (Lcom/amap/api/services/a/bx;, a), 
top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig), 
item: (Lcom/amap/api/services/a/bx;, a), 
method_set: [(Lcom/amap/api/services/a/bx;, a), (Lcom/amap/api/services/a/bx;, b), (Lcom/amap/api/services/a/j;, g)]

base_method: (Lcom/amap/api/services/a/bx;, a), 
top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig),
item: (Lcom/amap/api/services/a/bx;, b), 
method_set: [(Lcom/amap/api/services/a/bx;, a), (Lcom/amap/api/services/a/bx;, b), (Lcom/amap/api/services/a/j;, g)]

base_method: (Lcom/amap/api/services/a/bx;, b), 
top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig), 
item: (Lcom/amap/api/services/a/bx;, a), 
method_set: [(Lcom/amap/api/services/a/bx;, a), (Lcom/amap/api/services/a/bx;, b)]

base_method: (Lcom/amap/api/services/a/bx;, a), 
top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig), 
item: (Lcom/amap/api/services/a/bx;, a), 
method_set: [(Lcom/amap/api/services/a/bx;, a), (Lcom/amap/api/services/a/bx;, b), (Lcom/amap/api/services/a/j;, g)]

base_method: (Lcom/amap/api/services/a/bx;, a), 
top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig), 
item: (Lcom/amap/api/services/a/bx;, b), 
method_set: [(Lcom/amap/api/services/a/bx;, a), (Lcom/amap/api/services/a/bx;, b), (Lcom/amap/api/services/a/j;, g)]

base_method: (Lcom/amap/api/services/a/bx;, b), 
top_method: (Lcom/amap/location/offline/OfflineManager;, updateConfig), 
item: (Lcom/amap/api/services/a/bx;, a), 
method_set: [(Lcom/amap/api/services/a/bx;, a), (Lcom/amap/api/services/a/bx;, b)]
```

**Comment by krnick** [comment]
@sidra-asa Thanks for your help in testing, I will try to figure out a solution 😄
